rotkehlchen/chain/ethereum/modules/the_graph/decoder.py

```diff
@@ -0,0 +1,179 @@
+                        token=token,
+                    )
+                    event.notes += f', {taxed_amount} GRT has been burnt as delegation tax'
+                event.counterparty = CPT_THE_GRAPH
+        return DEFAULT_DECODING_OUTPUT
+
+    # processes StakeDelegatedLocked event
+    def _decode_stake_locked(
+            self,
+            token: 'EvmToken',
+            context: DecoderContext,
+    ) -> DecodingOutput:
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        tokens_amount = hex_or_bytes_to_int(context.tx_log.data[:32])
+        lock_timeout_secs = hex_or_bytes_to_int(context.tx_log.data[64:128])
+        tokens_amount_norm = token_normalized_value(
+            token_amount=tokens_amount,
+            token=token,
+        )
+        # create a new informational event about undelegation and the expiring lock on tokens
+        event = self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_GRT,
+            balance=Balance(),
+            location_label=delegator,
+            notes=(
+                f'Undelegated {tokens_amount_norm} GRT'
+                f' from Indexer {indexer}'
+                f', lock expires in {lock_timeout_secs} seconds'
+            ),
+            counterparty=CPT_THE_GRAPH,
+            address=context.tx_log.address,
+        )
+        return DecodingOutput(event=event)
+
+    # processes StakeDelegatedWithdrawn event
+    def _decode_stake_withdrawn(
+            self,
+            token: 'EvmToken',
+            context: DecoderContext,
+    ) -> DecodingOutput:
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        tokens_amount_norm = token_normalized_value(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[:32]),
+            token=token,
+        )
+        # create action item that will modify the relevant Transfer event that will appear later
+        action_item = ActionItem(
+            action='transform',
+            from_event_type=HistoryEventType.RECEIVE,
+            from_event_subtype=HistoryEventSubType.NONE,
+            asset=A_GRT,
+            amount=tokens_amount_norm,
+            to_event_type=HistoryEventType.STAKING,
+            to_event_subtype=HistoryEventSubType.REMOVE_ASSET,
+            to_notes=(
+                f'Withdrew {tokens_amount_norm} GRT'
+                f' from Indexer {indexer}'
+            ),
```

Comment:
Doesn' this fit in one line? If it's just for a tiny bit you can also add #noqa: E501
If they fit in one line keep them as such.

```suggestion
            to_notes=f'Withdrew {tokens_amount_norm} GRT from Indexer {indexer}',
```